client/src/js/modules/types/em/dc/redirect.vue

```diff
@@ -21,9 +21,25 @@ export default {
     },
     'computed': {
         redirectUrl: function () {
+            let redirectLocation = this.$store.state.appOptions.redirects.em;
             const visitStr = this.collection.queryParams.visit;
-            const [proposal, visit] = visitStr.split("-");
-            return `${this.$store.state.appOptions.redirects.em}/proposals/${proposal}/sessions/${visit}`
+
+            if (visitStr) {
+                const [proposal, visit] = visitStr.split("-");
+                redirectLocation += `/proposals/${proposal}/sessions/${visit}`;
+            } else {
+                const pathParams = window.location.pathname.split("/");
+                const lastParam = pathParams[pathParams.length - 1].split("-");
```

Comment:
If you don't need pathParams again, you can use pop to get the last item in the array. Not sure if it's better though!
```suggestion
                const lastParam = pathParams.pop().split("-");
```